### PR TITLE
Feature/gdpr  Confirm Opt In Email Sending Action Menu Item

### PR DESCRIPTION
**Pull request by gymad**
## Description

Adding a 'Send Confirm Opt In Email' action link to Accounts/Contacts/Leads/Prospects DetailViews action menu. New menu item shows only if email settings opt in sets to "Confirm Opt In".

Note: this PR contains a TODO but first have to be merged all task of GDPR to solve this, see: https://github.com/salesagility/SuiteCRM/compare/feature/gdpr...gymad:feature/gdpr_fixes_2_action_opt_in?expand=1#diff-f36992067482fe7579621b960419fe7fR383

## Motivation and Context

GDPR

## How To Test This

- QRR needed
- Try to send a Confirm Opt In Email via Action menu in Accounts/Contacts/Leads/Prospects DetailViews 
- Confirm Opt In email sending possible only if Email Settings / Opt In drop-down set to 'Confirm Opt In', try out with different settings, email sending should failing.
- Try out with incorrect email settings (for e.g. wrong email account / password) => email sending should failing, users have to get an error message about the email sending error. 
- Otherwise users have to get a success email sending message

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [x] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: include/MVC/View/SugarView.php [hunk 1]
```diff
@@ -766,10 +766,17 @@ public function displayHeader($retModTabs = false)
...
+            if (!empty($messages)) {
+                foreach ($messages as $message) {
+                    echo '<p class="error">' . $message . '</p>';
+                }
+            }
+            
+            $messages = SugarApplication::getSuccessMessages();
+            if (!empty($messages)) {
+                foreach ($messages as $message) {
+                    echo '<p class="success">' . $message . '</p>';
```
Comment (daniel-samson) [inline include/MVC/View/SugarView.php]: I :heart: this

File: include/language/en_us.lang.php [hunk 2]
```diff
@@ -2379,6 +2379,12 @@
     'LBL_SUITE_DESC2' => 'This program is free software; you can redistribute it and/or modify it under the terms of the GNU Affero General Public License version 3 as published by the Free Software Foundation, including the additional permission set forth in the source code header.',
     'LBL_SUITE_DESC3' => 'SuiteCRM is a trademark of SalesAgility Ltd. All other company and product names may be trademarks of the respective companies with which they are associated.',
     'LBL_GENERATE_PASSWORD_BUTTON_TITLE' => 'Reset Password',
+    'LBL_SEND_CONFIRM_OPT_IN_EMAIL' => 'Send Confirm Opt In Email',
+    'LBL_CONFIRM_OPT_IN_ONLY_FOR_PERSON' => 'Confirm Opt In Email sending only for Accounts/Contacts/Leads/Prospects',
+    'LBL_CONFIRM_OPT_IN_IS_DISABLED' => 'Confirm Opt In Email sending is disabled, enable Confirm Opt In option in Email Settings or contact your Administrator.',
+    'LBL_CONTACT_HAS_NO_PRIMARY_EMAIL' => 'Confirm Opt In Email sending is not possible because the Contact has not Primary Email Address',
+    'LBL_CONFIRM_EMAIL_SENDING_FAILED' => 'Confirm Opt In Email sennding failed',
```
Comment (horus68) [inline include/language/en_us.lang.php]: Typo in "sennding"

**Comment by Dillon-Brown** [comment]
Assessed

**Comment by Dillon-Brown** [comment]
Assessed

**Comment by Dillon-Brown** [comment]
Assessed

**Pull request merged**